adding period as special character

natapokie opened the pull request:
## **Contribution:**

Adding period as a special character. This was purely a frontend issue that validates text input.

### **Issue:**

Closes #546  

### **Accomplishments:**

- added period as special character

### **Visuals:** 

None

## **Details:**

### **Expected Behavior:**

- In previous versions, you would be unable to create an account (for example), if you used a period as a special character. An error message would show up since it didn't recognize the period
- Now, if you create an account, there would be no error message if you use a period as a special character in your password

### **Testing Steps:**

Steps required to get expected behavior.

- Only need to run frontend, `yarn dev`
- Create Account, and try to enter a password with a period "." as the special character, the button should be enabled and not disabled

### **Complications:**

- finding out how to use the match function, it seemed that we just need to add the period character into the square brackets []

### **Resources:**

Any additional resources used outside this in issue.

- [chatgpt](https://chat.openai.com/share/02d6ac1c-5ed0-4fd1-8ec0-24c7fb7f2cde) 😆

natapokie committed:
adding period as special character

Sherryzh0830 commented on client/src/pages/SignUp/functions.jsx:
```diff
@@ -11,7 +11,7 @@ export const validateEmail = (email) => {
 export function validatePassword(password) {
   return String(password).match(
-    '(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[@$!%*#?&])[A-Za-z0-9@$!%*#?&]{8,}',
+    '(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[.@$!%*#?&])[A-Za-z0-9@$!%*#?&.]{8,}',
```
Should we also account for other special characters here then as well? Such as ~ ` ^ + = ; and so on. I'm just pulling from this [website](https://www.computerhope.com/jargon/s/specchar.htm#:~:text=A%20special%20character%20is%20a,are%20examples%20of%20special%20characters.) for reference regarding what is a special character?

natapokie replied:
> Should we also account for other special characters here then as well? Such as ~ ` ^ + = ; and so on. I'm just pulling from this [website](https://www.computerhope.com/jargon/s/specchar.htm#:~:text=A%20special%20character%20is%20a,are%20examples%20of%20special%20characters.) for reference regarding what is a special character?

great point to mention! you can make a commit to add more of those characters :)

natapokie committed:
adding . to backend validation

natapokie committed:
reverting change to eslintrc

natapokie committed:
error message includes valid special characters

Sherryzh0830 committed:
Update functions.jsx to have other special characters

Special characters such as any type of brackets and quotation marks (both single and double quotation marks) are not accepted.

Sherryzh0830 committed:
Update SignUp.jsx (modified wording of the special charater requirements)

Sherryzh0830 committed:
Update UserServices.js to be consistent with the front end

Sherryzh0830 committed:
Forward/backward slashes
